components/rhs_thread/rhs_thread.jsx

```diff
@@ -100,7 +105,7 @@ export default class RhsThread extends React.Component {
         const curPostsArray = this.props.posts || [];
 
         if (this.props.socketConnectionStatus && !prevProps.socketConnectionStatus) {
-            this.props.actions.getPostThread(this.props.selected.id, false);
+            this.props.actions.getPostThread(this.props.selected.id, true);
```

Comment:
if you are enabling thread fetching here, then I don't see the reason for the rest of the code. this change will revert to old behavior and should just work...

Reply:
this part I added for a far fetched use case, when a user disconnects for long enough so there are enough posts in the middle to not have them all on the channel when he gets back.

But as I said, it is far fetched and can be reverted to false again if you feel it'll be better UX.

Reply:
My comment applies to line 95 as well. I think you can just leave line 95 and the rest of the changes are not needed.

Reply:
you are right, I forgot the websocket will load new posts even if they are not shown, which would happen if you follow a permalink, but I just checked and that's not a problem.

Anyways, this does not revert your changes completely, so the main channel benefits from not loading the whole threads. I do concurr that this should load progressively, but that should be a separate ticket

Reply:
@reflog why are we  reverting `this.props.actions.getPostThread(this.props.selected.id, true);`. I think this should be with `true` in the first place.

The condition `this.props.socketConnectionStatus && !prevProps.socketConnectionStatus` is present so as to load all the posts in the thread when socket reconnects as it can miss some posts because of disconnection.

Reply:
> I forgot the websocket will load new posts even if they are not shown

@Willyfrog yes, but it does only for the center channel but if we have thread from different channel open in RHS then it does not sync and hence this was added.